torchdata/datapipes/iter/util/plain_text_reader.py

```diff
@@ -58,10 +60,9 @@ def strip_newline(self, stream: Union[Iterator[bytes], Iterator[str]]) -> Union[
     def decode(self, stream: Union[Iterator[bytes], Iterator[str]]) -> Union[Iterator[bytes], Iterator[str]]:
         if not self._decode:
             yield from stream
-            return
```

Comment:
like the new control flow. Originally doesn't understand why there is `return` after `yield`...

Reply:
Yeah. I feel like the current implementation is easier to understand.
Regarding `return` and `yield`, as long as there is a `yield` in the function, the function is automatically classified as generator function no matter if `return` exists or not.